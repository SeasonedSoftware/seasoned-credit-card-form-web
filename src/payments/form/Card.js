import React from 'react'
import Cards from 'react-credit-cards'

export default props => {
  const { values, focused, acceptedCards, language } = props
  let cardProps = { ...values, focused, acceptedCards }
  cardProps = Object.assign(
    cardProps,
    language === 'pt-BR'
      ? {
          locale: { valid: 'valido at√©' },
          placeholders: { name: 'Seu nome' },
        }
      : {},
  )
  return <Cards {...cardProps} />
}
