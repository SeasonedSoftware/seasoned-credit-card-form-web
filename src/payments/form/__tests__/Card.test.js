import React from 'react'
import renderer from 'react-test-renderer'

import Card from '../Card'

jest.mock('react-credit-cards', () => props => <div {...props}> Cards </div>)

const props = {
  values: {},
  focused: '',
  acceptedCards: ['visa'],
}

describe('with en-US as language', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Card {...props} language={'en-US'} />)
    expect(tree).toMatchSnapshot()
  })
})

describe('with pt-BR as language', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Card {...props} language={'pt-BR'} />)
    expect(tree).toMatchSnapshot()
  })
})
