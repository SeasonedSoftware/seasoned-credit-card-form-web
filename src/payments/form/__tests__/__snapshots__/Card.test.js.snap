// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with en-US as language renders correctly 1`] = `
<div
  acceptedCards={
    Array [
      "visa",
    ]
  }
  focused=""
>
   Cards 
</div>
`;

exports[`with pt-BR as language renders correctly 1`] = `
<div
  acceptedCards={
    Array [
      "visa",
    ]
  }
  focused=""
  locale={
    Object {
      "valid": "valido atÃ©",
    }
  }
  placeholders={
    Object {
      "name": "Seu nome",
    }
  }
>
   Cards 
</div>
`;
